### PR TITLE
HOS-01 | Lack of Sanity Check - Remove Validator

### PR DESCRIPTION
**Pull request by asalzmann**
## What is the purpose of the change
When removing a validator with non-zero delegation amount from the set, those funds will be stranded until we re-add this validator.

## Brief Changelog

- ensure that Validator.DelegationAmt and Validator.Weight is ZERO before removing the target validator from a host zone.
- when rebalanding validators, sort ascending to avoid last-validator precision amount issue. See this example code: https://go.dev/play/p/3Zd_cVKZVyU

## Testing and verifying

CI + integration

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? audit

### CODE REVIEW

**Comment by asalzmann** [comment]
@riley-stride can you attach the go playground code used to test the validator sorting?

File: x/stakeibc/keeper/host_zone.go [hunk 1]
```diff
@@ -125,12 +125,17 @@ func (k Keeper) RemoveValidatorFromHostZone(ctx sdk.Context, chainId string, val
 	}
 	for i, val := range hostZone.Validators {
 		if val.GetAddress() == validatorAddress {
-			hostZone.Validators = append(hostZone.Validators[:i], hostZone.Validators[i+1:]...)
-			return true
+			if val.GetDelegationAmt() == 0 && val.GetWeight() == 0 {
+				hostZone.Validators = append(hostZone.Validators[:i], hostZone.Validators[i+1:]...)
+				return true
+			} else {
+				k.Logger(ctx).Error(fmt.Sprintf("Validator %s has non-zero delegation or weight", validatorAddress))
```
Comment (riley-stride) [inline x/stakeibc/keeper/host_zone.go:132]:
```suggestion
				k.Logger(ctx).Error(fmt.Sprintf("Validator %s has non-zero delegation (%d) or weight (%d)", validatorAddress, val.GetDelegationAmt(), val.GetWeight()))
```

**Review by riley-stride — APPROVED** [review]
LGTM
added my go/playground code to the PR description, pls take a look!
Also added one nit suggestion

**Comment by riley-stride** [comment]
Is this still a draft? @asalzmann

**Comment by riley-stride** [comment]
lgtm. fixed small merge conflict. let's merge @asalzmann !

**Pull request merged**